docs/book/src/developers/kubernetes-developers.md

```diff
@@ -35,13 +35,56 @@ export CLUSTER_TEMPLATE="test/dev/cluster-template-custom-builds.yaml"
 
 ## Testing the out-of-tree cloud provider
 
-To test changes made to the [Azure cloud provider](https://github.com/kubernetes-sigs/cloud-provider-azure), first build and push images for cloud-controller-manager and/or cloud-node-manager from the root of the cloud-provider-azure repo.
+To test changes made to the [Azure cloud provider](https://github.com/kubernetes-sigs/cloud-provider-azure), first build and push images for cloud-controller-manager and/or cloud-node-manager from the branch of the cloud-provider-azure repo that the desired changes are in. Based on the repository, image name, and image tag you produce from your custom image build and push, set the appropriate environment variables below:
```

Comment:
@CecileRobertMichon FYI a new set of documentation changes since your most recent review

Reply:
@nilo19 @lzhecheng could you please review this part?

Reply:
lgtm

Reply:
lgtm. Thanks!